packages/store-sync/src/blockLogsToStorage.ts

```diff
@@ -0,0 +1,305 @@
+import {
+  decodeField,
+  decodeKeyTuple,
+  decodeRecord,
+  hexToTableSchema,
+  abiTypesToSchema,
+  TableSchema,
+} from "@latticexyz/protocol-parser";
+import { GroupLogsByBlockNumberResult, GetLogsResult } from "@latticexyz/block-logs-stream";
+import { StoreEventsAbi, StoreConfig } from "@latticexyz/store";
+import { TableId } from "@latticexyz/common";
+import { Address, Hex, decodeAbiParameters, parseAbiParameters } from "viem";
+import { debug } from "./debug";
+// TODO: move these type helpers into store?
+import { Key, Value } from "@latticexyz/store-cache";
+import { isDefined } from "@latticexyz/common/utils";
+import { SchemaAbiType, StaticAbiType } from "@latticexyz/schema-type";
+
+// TODO: change table schema/metadata APIs once we get both schema and field names in the same event
+
+// TODO: export these from store or world
+export const schemaTableId = new TableId("mudstore", "schema");
+export const metadataTableId = new TableId("mudstore", "StoreMetadata");
+
+// I don't love carrying all these types through. Ideally this should be the shape of the thing we want, rather than the specific return type from a function.
+export type StoreEventsLog = GetLogsResult<StoreEventsAbi>[number];
+export type BlockLogs = GroupLogsByBlockNumberResult<StoreEventsLog>[number];
```

Comment:
could you elaborate on this comment? having trouble following

Reply:
I was thinking that using the `GetLogsResult` import type ties it the shape defined in the other module (and its package version) and was worried it make it harder to pass in an object from another location (rather from this specific function in this specific module). But maybe that's not how these types actually work and they just compare shapes.

This worry comes from seeing weird type errors when certain package versions get out of alignment and is why we have most deps pinned to a specific version rather than a version range.